### DISCUSSION TITLE
For people encountering ssl.SSLError: [ASN1] nested asn1 error (_ssl.c:4145)

### DISCUSSION THREAD

**Discussion by levicki** [hlky/stable-diffusion-webui#1325] [General]
**Cause:**
Python bug with loading improperly encoded CA certificates from Windows root store:
https://bugs.python.org/issue35665

**Resolution:**
Change your `Miniconda3/envs/ldm/lib/ssl.py` code as follows:
```
    def _load_windows_store_certs(self, storename, purpose):
        certs = bytearray()
        try:
            for cert, encoding, trust in enum_certificates(storename):
                # CA certs are never PKCS#7 encoded
                if encoding == "x509_asn":
                    if trust is True or purpose.oid in trust:
                        try:
                            self.load_verify_locations(cadata=cert)
                            certs.extend(cert)
                        except SSLError:
                            warnings.warn("Bad certificate in Windows certificate store")
        except PermissionError:
            warnings.warn("unable to enumerate Windows certificate store")
        return certs
```

**Explanation:**
Code change will make Python verify Windows root store CA certificates one by one and only throw a warning on invalid ones instead of crashing.

**Comment by nmirkov** [comment]
This helped to resolve my issue, I also added logging to locate the bad certificate (of course it was a government issued one :) ).
Thanks a lot!

**Comment by IgorCode** [comment]
I assume the issue is from one of the MUP Srbija certificates?

**Reply by Vihershunde** [reply]
Ne mogu da verujem hahaha 🤣

**Reply by Grbson00** [reply]
Neee je l moguce, 2 sata pokusavam da resim i naravno da je MUP

**Reply by MarkoStanojevic12** [reply]
Jbt ja ne mogu da verujem na sta sam izgubio vreme...

**Reply by horosavin** [reply]
Momci, izgleda da me neki od ovih sertifikata sprecava da instaliram Tydiffusion u 3DS Max. Instalacija koristi "embedded" pitona i uopste ne znam u koji bi ssl (ima ih nekoliko u instalacionom folderu) trebalo da modifikujem i kako - posto nisam programer.
Da li imate neko resenje?

**Reply by IgorCode** [reply]
I "_load_windows_store_certs" se nalazi u svim tim ssl fajlovima?

**Reply by horosavin** [reply]
Zapravo, ne nalazi se ni u jednom...

**Reply by horosavin** [reply]
Karambol je napravio  24x07.co.yu Digitrust CA. Asseco. Sta reci...

**Reply by sw-dev-code** [reply]
@IgorCode , @horosavin, @Vihershunde  Mozete li mi reci kako ste resili problem? Izgleda da je isti slucaj kod mene

**Reply by IgorCode** [reply]
Izmenom `_load_windows_store_certs` funcije kao što je gore i napisano.

**Comment by Vihershunde** [comment]
Thank you so much, it fixed the bug.
The issue is the Serbian government's certificates.

**Comment by yiqiangjun** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by whiteleon13** [comment]
Anyone who is having problems to even find ssl.py: get your python executable location, in that Python folder there is ssl.py and the function listed up

**Comment by yiqiangjun** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by yiqiangjun** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
